test_runner/batch_others/test_wal_acceptor_async.py

```diff
@@ -109,7 +109,7 @@ async def wait_for_lsn(safekeeper: Safekeeper,
                        timeline_id: str,
                        wait_lsn: str,
                        polling_interval=1,
-                       timeout=600):
+                       timeout=200):
```

Comment:
`wait_for_lsn`'s timeout is for a single safekeeper for a single iteration only. Six iterations times two safekeepers (one is a victim) times 200s timeout is 40 minutes per test. Seems excessive to me, and it's definitely much bigger than the `no_output_timeout`, so it probably won't help.

E.g. in https://app.circleci.com/pipelines/github/zenithdb/zenith/3061/workflows/42b1e5ca-db66-41aa-9284-f9b3059ae060/jobs/24588/steps `test_restarts_under_load` starts at `2021-11-03 10:25:38.251`, followed by `waiting for lsn` at:

* `2021-11-03 10:25:44.213` and `2021-11-03 10:25:44.217`, synced at `2021-11-03 10:25:54.218`
* `2021-11-03 10:25:55.309` and `2021-11-03 10:26:04.344`, synced at `2021-11-03 10:26:14.345`
* `2021-11-03 10:26:15.424` and `2021-11-03 10:26:15.428`, synced at `2021-11-03 10:26:45.501`
* `2021-11-03 10:26:46.585` and `2021-11-03 10:26:46.589`, synced at `2021-11-03 10:27:19.670`
* `2021-11-03 10:27:20.775` and `2021-11-03 10:27:40.882`, synced at `2021-11-03 10:27:50.883`
* `2021-11-03 10:27:51.987` and `2021-11-03 10:27:51.992`, synced at `2021-11-03 10:28:22.69`

It seems that each iteration takes between 10 and 30 seconds, so I'd use actually lower timeout to, say, 45 or 60 seconds. In the latter case the maximum time spent in a test is about 12 minutes, so raising `no_output_timeout` starts looking like a good idea.

Looking at the logs, maybe timeout should actually be per-iteration, not per-safekeeper.

Reply:
> synced at HH:MM:SS

Hmm, I think you're using `All workers made N transactions` entry as a "synced at" moment, but sync actually completes 10 seconds before that, there's `asyncio.sleep(10)` in the code after sync.

> Looking at the logs, maybe timeout should actually be per-iteration, not per-safekeeper.

It's actually not that different. In single iteration we have 2 alive safekeepers, where one has just been started and another has been working for some time. Safekeeper which was working in previous iteration should have all WAL up to wait_lsn and we won't wait for him. Another safekeeper (which has just been started) is likely to fall behind significantly and we will spend most time waiting for him in `wait_for_lsn`.

> It seems that each iteration takes between 10 and 30 seconds

I would say this duration is very unstable. When I was adjusting timeout last time, I tried to set timeout=120 and it didn't work, because sync was too slow to complete in 120 seconds, so I set it to 600. From my recent observations, wal_proposer sync speed is very unstable, and can be slow and fast in the two consecutive test runs.

In my opinion, this unstable behavior is the root cause of timeout errors and I hope we will fix it soon. It seems that syncing always took about ~6-10 seconds when everything was stable. I would say this PR tries to mitigate CircleCI timeouts temporarily, until root cause is fixed. CircleCI timeouts are not very good because they prevent pytest logs from being shown.